File: resources/capability-bundle/.claude/skills/profile-evidence/scripts/check-profile-candidate.py
import json
import sys


candidate = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
states = {
    "confirmed_fact",
    "learner_viewpoint",
    "observation",
    "hypothesis",
    "conflicted",
    "retired",
}

errors = []
if candidate.get("state") not in states:
    errors.append("invalid assertion state")
if not str(candidate.get("claim", "")).strip():
    errors.append("claim is required")
if not isinstance(candidate.get("evidence_refs"), list) or not candidate["evidence_refs"]:
    errors.append("at least one evidence reference is required")
if not str(candidate.get("scope", "")).strip():
    errors.append("scope is required")

sys.stdout.write(json.dumps({"valid": not errors, "errors": errors}, separators=(",", ":")))

File: resources/capability-bundle/.claude/skills/profile-evidence/scripts/check-profile-candidate.mjs
const candidate = JSON.parse(process.argv[2] ?? '{}');
const states = new Set([
  'confirmed_fact',
  'learner_viewpoint',
  'observation',
  'hypothesis',
  'conflicted',
  'retired',
]);

const errors = [];
if (!states.has(candidate.state)) errors.push('invalid assertion state');
if (!String(candidate.claim ?? '').trim()) errors.push('claim is required');
if (!Array.isArray(candidate.evidence_refs) || candidate.evidence_refs.length === 0) {
  errors.push('at least one evidence reference is required');
}
if (!String(candidate.scope ?? '').trim()) errors.push('scope is required');

process.stdout.write(JSON.stringify({ valid: errors.length === 0, errors }));
